[Quality Declaration] Fixed rep links and added more details to dependencies (#1116)
ros2/rclcpp: rclcpp_lifecycle/QUALITY_DECLARATION.md

## Dependencies [5]

Below are evaluations of each of `rclcpp_lifecycle`'s run-time and build-time dependencies that have been determined to influence the quality.

> It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.

Reviewer: ```suggestion
It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.

```

It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.

### Direct and Optional Runtime ROS Dependencies [5.i]/[5.ii]

`rclcpp_lifecycle` has the following runtime ROS dependencies:

#### `lifecycle_msgs`

> The `lifecycle_msgs` package contains some lifecycle related message and service definitions.

Reviewer: ```suggestion
The `lifecycle_msgs` package contains message and service definitions for managing lifecycle nodes.
```

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rcl_interfaces/blob/master/lifecycle_msgs/QUALITY_DECLARATION.md).

#### `rclcpp`

The `rclcpp` package provides the ROS client library in C++.

It is **Quality Level 4**, see its [Quality Declaration document](https://github.com/ros2/rclcpp/blob/master/rclcpp/QUALITY_DECLARATION.md).

#### `rcl_lifecycle`

> The `rcl_lifecycle` package provides a C-based lifecycle implementation.

Reviewer: ```suggestion
The `rcl_lifecycle` package provides functionality for ROS 2 lifecycle nodes in C.
```
